agentMET4FOF/metrological_agents.py

```diff
@@ -192,3 +193,118 @@ def custom_plot_function(self, data, sender_agent, **kwargs):
+        self.supported_datatypes.append(TimeSeriesBuffer)
+
+    def convert_single_to_tsbuffer(self, single_data: Union[List, Tuple, np.ndarray]):
+        """Convert common data in agentMET4FOF to :class:`TimeSeriesBuffer
+        <time-series-buffer:time_series_buffer.buffer.TimeSeriesBuffer>`
+
+        Parameters
+        ----------
+        single_data : iterable of iterables (list, tuple, np.ndarrray) with shape (N, M)
+
+            * M==2 (pairs): assumed to be like (time, value)
+            * M==3 (triple): assumed to be like (time, value, value_unc)
+            * M==4 (4-tuple): assumed to be like (time, time_unc, value, value_unc)
+
+        Returns
+        -------
+        TimeSeriesBuffer
+            the new :class:`TimeSeriesBuffer
+            <time-series-buffer:time_series_buffer.buffer.TimeSeriesBuffer>` object
+
+        """
+        ts = TimeSeriesBuffer(maxlen=self.buffer_size)
+        ts.add(single_data)
+        return ts
+
+    def update(
+            self,
+            agent_from: str,
+            data: Union[Dict, List, Tuple, np.ndarray],
+    ) -> TimeSeriesBuffer:
+        """Overrides data in the buffer dict keyed by `agent_from` with value `data`
+
+        Parameters
+        ----------
+        agent_from : str
+            Name of agent sender
+        data : dict or iterable of iterables (list, tuple, np.ndarray) with shape (N, M
+            the data to be stored in the metrological buffer
+
+        Returns
+        -------
+        TimeSeriesBuffer
+            the updated :class:`TimeSeriesBuffer
+            <time-series-buffer:time_series_buffer.buffer.TimeSeriesBuffer>` object
+        """
+        # handle if data type nested in dict
+        if isinstance(data, dict):
+            # check for each value datatype
+            for key, value in data.items():
+                data[key] = self.convert_single_to_tsbuffer(value)
+        else:
+            data = self.convert_single_to_tsbuffer(data)
+            self.buffer.update({agent_from: data})
+        return self.buffer
+
+    def store(self, agent_from, data=None, concat_axis=0):
+        # if first argument is the agentMET4FOF dict message
+        if isinstance(agent_from, dict):
+            message = agent_from
+        # otherwise, we expect the name of agent_sender and the data to be passed
+        else:
+            message = {"from": agent_from, "data": data}
+
+        # store into a separate variables, it will be used frequently later for the type checks
+        message_from = message["from"]
+        message_data = message["data"]
+
+        # check if sender agent has sent any message before:
+        # if it did,then append, otherwise create new entry for the input agent
+        if message_from not in self.buffer:
+            self.update(message_from, message_data)
+            return 0
+        else:
+            self._concatenate(iterable=self.buffer[message_from],data=message_data, concat_axis=concat_axis)
```

Comment:
I am pretty sure we do not need this. After overwriting `_concatenate` the base `store` should work, I think.

Reply:
@bangxiangyong Do you see a reason, to leave it in?

Reply:
@bangxiangyong Before we merge this, do you see a reason to leave this store in? I think it is redundant and not really needed.

Reply:
@bangxiangyong I think we can drop the overwriding of the `store` method. Do you agree?